src/frontends/tensorflow_common/src/op/crop_and_resize.cpp

```diff
@@ -3,137 +3,73 @@
-            } else {  // nearest
-                interpolate_attrs.mode = Interpolate::InterpolateMode::NEAREST;
-            }
-
-            ng_crop = make_transpose(ng_crop, {0, 3, 1, 2});
-            auto ng_output =
-                make_shared<Interpolate>(ng_crop, ng_size, ng_scales, ng_axes, interpolate_attrs)->output(0);
-            ng_output = make_transpose(ng_output, {0, 2, 3, 1});
-            ng_crop_outputs.at(i) = ng_output;
-        }
+OutputVector translate_crop_and_resize_bilinear(const NodeContext& node) {
+    default_op_checks(node, 4, {"CropAndResize"});
+    auto image = node.get_input(0);
+    auto boxes = node.get_input(1);
+    auto box_ind = node.get_input(2);
+    auto crop_size = node.get_input(3);
+
+    // try to retrive crop_size as the constant, otherwise, we cannot support it
+    Shape crop_sizes;
+    get_const_input(node, 3, &crop_sizes);
+
+    // concatenate boxes and box_ind inputs because
+    // ROIPooling accepts ROIs in a format [batch_id, x_1, y_1, x_2, y_2]
+    // prepare box_ind for futher concatenation
+    auto const_one = make_shared<Constant>(element::i32, Shape{1}, 1);
+    box_ind = make_shared<Unsqueeze>(box_ind, const_one);
+    box_ind = make_shared<Convert>(box_ind, element::f32);
+    boxes = make_shared<Concat>(OutputVector{box_ind, boxes}, 1);
+
+    // boxes are going in the format [y1, x1, y2, x2]
+    // so we need to adjust them to the format [x_1, y_1, x_2, y_2]
+    // use Gather operation for the swapping
+    auto gather_order = make_shared<Constant>(element::i32, Shape{5}, vector<int32_t>{0, 2, 1, 4, 3});
+    auto gather_axis = make_shared<Constant>(element::i32, Shape{1}, vector<int32_t>{1});
+    boxes = make_shared<Gather>(boxes, gather_order, gather_axis);
+
+    // prepare input image for ROIPooling
+    image = make_transpose(image, {0, 3, 1, 2})->output(0);
+    Output<Node> roi_pooling = make_shared<ROIPooling>(image, boxes, crop_sizes, 1.0f, "bilinear");
+    roi_pooling = make_transpose(roi_pooling, {0, 2, 3, 1})->output(0);
+    set_node_name(node.get_name(), roi_pooling.get_node_shared_ptr());
+    return {roi_pooling};
+}
 
-        auto res = make_shared<Concat>(ng_crop_outputs, 0);
-        set_node_name(node.get_name(), res);
-        return res->outputs();
-    }
+OutputVector translate_crop_and_resize_op(const NodeContext& node) {
+    // retieve attributes
+    // some cases like non-zero extrapolation_value are not supported
+    auto method = node.get_attribute<string>("method", "bilinear");
+    auto extrapolation_value = node.get_attribute<float>("extrapolation_value", 0.0f);
+
+    TENSORFLOW_OP_VALIDATION(
+        node,
+        method == "nearest" || method == "bilinear",
+        "[TensorFlow Frontend] Inconsistent model: CropAndResize support only bilinear and nearest sampling methods.");
+
+    // TODO 102603: handle a case with non-zero extrapolation_value and nearest sampling
+    TENSORFLOW_OP_VALIDATION(
```

Comment:
did the previous implementation support "nearest" mode or extrapolation_value != 0?

Reply:
No, nothing supported:) No layer test is passing:(
I implemented it from scratch. Moreover, we have no models requiring `nearest` sampling method. There is only Mask R-CNN with bilinear method. Now it is converted.